staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go

```diff
@@ -131,6 +131,9 @@ func (a *MutatingWebhook) SetServiceResolver(sr config.ServiceResolver) {
 // SetScheme sets a serializer(NegotiatedSerializer) which is derived from the scheme
 func (a *MutatingWebhook) SetScheme(scheme *runtime.Scheme) {
 	if scheme != nil {
+		// TODO: we should not mutate a scheme we don't own. We should have a fallback serializer instead.
+		admissionv1beta1.AddToScheme(scheme)
+
 		a.clientManager.SetNegotiatedSerializer(serializer.NegotiatedSerializerWrapper(runtime.SerializerInfo{
 			Serializer: serializer.NewCodecFactory(scheme).LegacyCodec(admissionv1beta1.SchemeGroupVersion),
```

Comment:
Maybe we can create a scheme that only understands admission/v1beta1 and use it to create a serializer. I'm trying it locally.

Reply:
We have RawExtensions. Won't work. With some more effort we can get it to work. @hzxuzhonghu is on it. I would prefer an easy fix here for 1.10 and 1.9.

Reply:
The RawExtensions uses `json.Marshal` to encode: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/extension.go#L38-L48

I tested locally it works fine.

Reply:
Interesting about json.Marshal. This should really be an encoder.

Why do we pass the scheme at all into the plugin then?

Reply:
We need it for conversion.

Reply:
Yeah, for conversion.

> Interesting about json.Marshal. This should really be an encoder.

Since the object is already converted, we really only need to serialize it. A custom json deserializer has better performance, but for the `json.Marshal` is alright for serializing.

We need to revisit when apiserver communicates with webhooks in proto.